constant addresses autocomplete ts

greg-schrammel opened the pull request:
makes constant addresses autocomplete
![Screen Shot 2021-11-29 at 23 42 58](https://user-images.githubusercontent.com/6232729/143976335-6c56116f-a04e-4134-8d9b-2764be811615.png)

Commit:
constant addresses autocomplete ts

Commit:
constant addresses autocomplete ts

0xdavinchee commented on src/slices/AccountSlice.ts:
```diff
@@ -50,18 +50,24 @@ export const getBalances = createAsyncThunk(
+          // fsohmContract.signer;
+          return fsohmContract.callStatic.balanceOfUnderlying(address);
+        }),
+      ).then(underlyingBalances =>
```
personally not the biggest fan of `.then`, but this is fine

greg-schrammel replied:
I'm not sure why I refactored this, maybe to use promise.all but what do you think of just wrapping the old `for` loop with an `if` and call it a day

ohmygirth committed:
Avax migrate (#778)

* Add basic top bar button to open ChangeNetwork modal view

* Update web3Context to include Arbitrum and provide chainName value

* Add Arbitrum URI to web3Context

* Add change network buttons for eth, arbitrum, calling stub function in web3context

* adding constants for ohm-weth arbitrum contract setup.

* commented Bond.ts for helpful future cleanup (architecture-wise).

* Send request to switch chain

* Request permission to add network to wallet if attempting to switch to one the user hasn't added

* Close the ChangeNetwork modal after submitting the request

* Use Box for network list layout

* refactored to create a new interface ArbitrumBond, so regular StableBonds don't need ghost Arbitrum addresses no more.

* deleted unecessary comment.

* reverted back to NetworkAddresses having all 4 networkId props after unbasky feedback.

* changed ohm_weth from LPBond to CustomBond.

* reverted back ohm_weth to LPBond.

* merged updated contracts changes onto initial arbitrum changes branch. awaiting actual contract values to update variables.

* Fix styling bug causing NetworkMenu button to be narrower than the other buttons in TopBar

* Add block explorer urls for arbitrum & arbitrum-testnet to ConnectMenu getEtherscanUrl function

* Move NetworkMenu's logic for getting chain name into a hook, display on ChangeNetwork modal too

* Replace ChangeNetwork Box layout with Grid, enlarge buttons

* Remove GetCurrentChain, add improved messaging in ChangeNetwork when connected to an unsupported chain. Add arbitrum testnet to web3context

* Add logos to buttons, change to column format for narrow screens

* Add logos to NetworkMenu button

* Get stats from contract/coingecko instead of graphData when on arbitrum

* Fix alchemy api uris

* Fix typo

* Restyle ChangeNetwork according to 0xmachin4's figma examples

* Change to flex columns in mobile mode

* Update top bar button - rounded image, improve padding

* Highlight currently selected network button

* Fix so button fills entire grid parent

* Highlight currently active network name

* minor refactors on NetworkMenu.jsx and ChangeMenu.jsx

* personal environment prettier change to run project

* removed 'message' state related code from ChangeNetwork.jsx.

* fixed text positioning for Network Modal on mobile devices.

* removed unused import

* More layout fixes - button centering on narrow mobile screens, header font weight, narrower gap between button icons and text, remove unnecessary css & unused classes

* Update top bar ConnectMenu button to use same font size, weight and button padding as OhmMenu and NetworkMenu

* With the extra NetworkMenu button, the top bar can become too long to fit on narrow screens, especially if a wallet is not connected so "Connect Wallet" is displayed rather than "Disconnect". So, on mobile hide text for OhmMenu and NetworkMenu, only displaying the icons.

* Merge branch 'develop' into choose-network-modal
Environment.ts and web3Context.tsx updated to integrate arbitrum env config

# Conflicts:
#	src/themes/dark.js
#	src/themes/light.js

* Update .env.example with changes to support arbitrum

* Remove duplicate lusd bond; merge mistake

* Update Environment.ts and .env.example to account for eth/arb self hosted websockets

* Add tidied up version of NETWORKS back to constants.ts, required to add new chains to the user's wallet

* Add placeholder network ids for arbitrum

* Update NodeHelper to account for multiple chains

* Hide/show available bonds according to network

* Set provider on connect regardless of network validation - otherwise cannot change from unsupported chain to eth/arb

* Make bond networkAddrs optional, add checks to places expecting them, hide ExternalStakePool if ohm-lusd is unavailable on current network

* Hide 3,3 together when not on eth or rinkeby

* If not on eth or rinkeby /33-together route redirects to /stake

* Set chainID in web3Context even if chain is invalid - prevents making calls to eth apis and loading data as if we are on eth when on unsupported chains

* Add payout token to Bond, display on BondPurchase modal instead of hardcoded "OHM"

* refactored nested ternary operators for rendering views into separate expressions.

* when arbitrum chain is selected, rendering message explaining user cannot stake.

* Merge branch 'develop' into choose-network-modal

# Conflicts:
#	src/App.jsx
#	src/helpers/AllBonds.ts
#	src/hooks/useGoogleAnalytics.ts
#	src/hooks/web3Context.tsx
#	src/slices/BondSlice.ts

* Remove obsolete comment and unused const

* Replace old ethereum.png with wETH.svg

* Left justify button content on change network modal

* Update styles for selected button

* Update modal layout

* refactored nested ternaries operators for rendering views into separate expressions | when Arbitrum chain is selected, no stacking message rendered.

* prettier/prettier changes

* Change staking unavailable message

* Merge latest develop

* Move logic into NetworkSlice

* Add avax

* Add missing avax stuff to AllBonds.tx

* Add address lists for avax

* Add missing address from web3context

* Improvements to network switching

* adds network slice and avax migrator contracts

* merge & update env node vars

* try/catch loadAccount & getBalance errors that were preventing return

* new gohm address, removed network logic on routes. approval broke, balances broke

* fixed broke migrate, balance still doesnt pull

* will load for unconnected wallet now

* using static provider on avax appslice

* moving wsohm up higher in the try/catch

* removing extra package-lock, merged in somewhere previously

* uris werent returning a string

* moved initial modal

* fix wrap state on initial connect

* put back fiatdao address

Co-authored-by: spaceturtleship <spaceturtleship@protonmail.com>
Co-authored-by: Joao Tavares <jc.992@hotmail.com>
Co-authored-by: João Tavares <joaot@LAPTOP-MIBU954S>
Co-authored-by: Joao Tavares <joao.tavares@twistag.com>
Co-authored-by: appleseed-iii <appleseed-iii@protonmail.com>

ohmygirth committed:
commented out conditional in network slice (#781)

* commented out conditional

* moved try down

appleseed-iii committed:
update gohm & fiatdao wsohm calcs (#782)

ohmygirth committed:
fixes add chain to wallet  (#783)

* commented out conditional

* moved try down

* fixed

Commit:
I don't even know anymore

greg-schrammel commented:
closing since this requires bigger changes than expected and should be tacked from a different angle
